### ISSUE TITLE
8$: Lint use-before-decl for TypeScript import aliases

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#793]
The following should report an error:
```typescript
namespace ns { export class B {} }

new A();  // use before decl
import A = ns.B;
```
